packages/docusaurus/src/server/html-tags/htmlTags.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import _ from 'lodash';
+import {HtmlTagObject} from '@docusaurus/types';
+import htmlTags from 'html-tags';
+import voidHtmlTags from 'html-tags/void';
+
+function assertIsHtmlTagObject(val: any): asserts val is HtmlTagObject {
+  if (!_.isPlainObject(val)) {
+    throw new Error(`"${val}" is not a valid HTML tag object`);
+  }
+  if (typeof val.tagName !== 'string') {
+    throw new Error(
+      `${JSON.stringify(
+        val,
+      )} is not a valid HTML tag object. "tagName" must be defined as a string`,
+    );
+  }
+}
+
+export function htmlTagObjectToString(tagDefinition: any): string {
```

Comment:
Can we find an open source equivalent for this? I'm not sure if writing a serializer ourselves is the best idea.

Reply:
Ah i forgot to mention that this is a popular API from https://github.com/jantimon/html-webpack-plugin which has 2,.2million

In fact, its taken from there. But not `xhtml` compliant
https://github.com/jantimon/html-webpack-plugin/blob/bf017f31e29161bdcafb03cff90fb635ed54438a/lib/html-tags.js#L22-L44

Reply:
Oh I see. I've used html-webpack-plugin before but didn't know they supported such an API. WDYT about the string-based? That can be the fallback in case we don't support something. Lol.